### ISSUE TITLE
Printing labels sometimes fails with "Error ... returned HTTP code: 0"

### ISSUE THREAD

**Issue by melek** [Automattic/woocommerce-services#2360]
**The issue**
A user is intermittently getting 'HTTP 0' errors when creating labels (28267948-hc → 3803849-zen)

These are the two errors the user sees:
- Error: The WooCommerce Shipping & Tax server returned HTTP code: 0
- Failed to retrieve shipping label refund status: Error: The WooCommerce Shipping & Tax server returned HTTP code: 0

User tried conflict testing with Health Check, the error recurs with a default theme and only WooCommerce, Jetpack, and WooCommerce Shipping & Taxes enabled.

Will update here with more information from SSR once the user provides it to us.

**Additional Notes**
* Started happening suddenly around beginning of March.
* This does not happen every order. 
* User can reload an order and then purchase a label, but then it will create 2 labels and the user has to request a refund for one of them.
* Similar issue reported in Slack (p1614727285128500-slack-C7U3Y3VMY)

**Comment by aheckler** [comment]
3790710-zen is this as well. @Ferdev they are on WC 5.0.0 and WC Shipping + Tax 1.25.8.

There is another ticket (3765612-zen) that had this issue as well, but it was solved by conflict testing, so I'm going to have 3790710-zen conflict test too.

**Comment by johndcoy** [comment]
Another report of this 3779293-zen

**Comment by tijosh** [comment]
@c-shultz all the sites reporting this problem so far share the same host (GD). Is this possibly an IP blocking issue?

**Comment by c-shultz** [comment]
@tijosh 

I don't see any reason to expect an IP blocking issue on our end. We have made specific requests in the past to block certain IPs that were hammering our server, but I don't see any history of that for any of these sites. The `cURL error 28` is indicating a communication problem between their site and our server (`api.woocommerce.com`). This could be happening if our server wasn't responding to them soon enough (if their IP was blocked by us or our server was otherwise down), or it could be happening if there was a problem with their hosting. Could we ask them to check with their host to determine if there could be something block their site from accessing `api.woocommerce.com`?

**Comment by AashikP** [comment]
Thank you, @c-shultz. I'm currently checking with them if updating their PHP version and cURL versions makes a difference.

**Comment by nicholasjeberlin** [comment]
34371712-hc

**Comment by FangedParakeet** [comment]
Added to the Fulfill Codeable query queue.

**Comment by dustinparker** [comment]
@nicholasjeberlin is this still an ongoing issue with 34371712-hc?

**Comment by nicholasjeberlin** [comment]
Hi @dustinparker,

Sorry for just seeing this. I've created a follow-up to inquire if their client is still running into the issue 5219500-zen.

**Comment by thisissandip** [comment]
Another: 5220024-zen

Asked them to test using the recent version of the plugin

**Comment by thisissandip** [comment]
Another one: 5218847-zen

Host: BlueHost

Error logs:

```
05-16-2022 @ 07:26:27 - wcc_server_error Error: The WooCommerce Shipping & Tax server returned HTTP code: 0 (fetch_payment_methods_from_connect_server)
05-16-2022 @ 07:27:29 - wcc_server_error Error: The WooCommerce Shipping & Tax server returned HTTP code: 0 (WC_REST_Connect_Shipping_Label_Status_Controller)
05-16-2022 @ 07:30:49 - wcc_server_error Error: The WooCommerce Shipping & Tax server returned HTTP code: 0 (WC_REST_Connect_Shipping_Label_Status_Controller)
05-16-2022 @ 07:31:19 - wcc_server_error Error: The WooCommerce Shipping & Tax server returned HTTP code: 0 (fetch_payment_methods_from_connect_server)
05-16-2022 @ 07:51:18 - wcc_server_error Error: The WooCommerce Shipping & Tax server returned HTTP code: 0 (WC_REST_Connect_Shipping_Label_Controller)
05-16-2022 @ 10:16:42 - wcc_server_error Error: The WooCommerce Shipping & Tax server returned HTTP code: 0 (WC_REST_Connect_Shipping_Rates_Controller)
```

I found the curl 28 error on JP debug:

```
rest_api_endpoint_with_blog_token
http_request_failed: cURL error 28: Connection timed out after 5000 milliseconds

xmlrpc_endpoint_with_user_token
ixr_-10520: Jetpack: [http_request_failed] cURL error 28: Connection timeout after 10000 ms

rest_api_endpoint_with_user_token
http_request_failed: cURL error 28: Connection timed out after 5000 milliseconds
```

I have asked them to reset JP connection. @dustinparker can you please check this and let me know what should be the next step?

---

Also, the users from the above tickets ( 3790710-zen & 3765612-zen ) followed up and mentioned that conflict testing does not solve the issue.

**Comment by thisissandip** [comment]
Looks like this issue is being handled by JPOP: p1652793128502319-slack-CB37YNEH1

**Comment by dustinparker** [comment]
@thisissandip is the curl error happening every time on those 4 sites, or just occasionally? Are they able to get label rates and print labels at all? Are they all on Bluehost?

Could you share the WC system reports for the 4 tickets you mentioned? I don't have access to the tickets so can't see what the merchants have shared.

**Comment by dustinparker** [comment]
hi @thisissandip, just wanted to follow up and see if we can get the information requested in [the last comment](https://github.com/Automattic/woocommerce-services/issues/2360#issuecomment-1131238519). Please let me know when you get a chance, thanks!

**Comment by thisissandip** [comment]
Hey @dustinparker, thanks for following up! It turned out to be an issue with Bluehost blocking Jetpack's IP. The merchants from the above tickets followed up that this is now fixed.

**Comment by dustinparker** [comment]
ok, glad to hear it! Thanks for the update. I'll go ahead and close this out as it seems that it is not an issue with the WC connect server itself.
